packages/library-sync/src/commonMain/kotlin/io/realm/kotlin/mongodb/internal/RealmSyncUtils.kt

```diff
@@ -79,6 +80,7 @@ internal fun convertSyncError(syncError: SyncError): SyncException {
     val errorCode = syncError.errorCode
     val message = createMessageFromSyncError(errorCode)
     return when (errorCode.errorCode) {
+        ErrorCode.RLM_ERR_AUTH_ERROR -> AuthExpiredException(message)
```

Comment:
We should also send this to the AuthenticationListener

Reply:
We should probably update the `AuthenticationListener` to take advantage of https://github.com/realm/realm-core/pull/7302 instead of issuing these events ourselves.

Reply:
> We should also send this to the AuthenticationListener

We should probably update the `AuthenticationListener` to take advantage of https://github.com/realm/realm-core/pull/7302 instead of issuing these events ourselves.